queries.sql

```diff
@@ -98,4 +98,37 @@ SELECT a.name, a.escape_attempts, o.full_name FROM animals AS a INNER JOIN owner
 
 
 -- Who owns the most animals?
-SELECT COUNT(a.name) AS anim_qty, o.full_name FROM animals AS a INNER JOIN owners AS o ON a.owners_id = o.id GROUP BY o.full_name ORDER BY anim_qty DESC LIMIT 1;
\ No newline at end of file
+SELECT COUNT(a.name) AS anim_qty, o.full_name FROM animals AS a INNER JOIN owners AS o ON a.owners_id = o.id GROUP BY o.full_name ORDER BY anim_qty DESC LIMIT 1;
+
+
+
+----------------------Day-4-Project--------------------------------
+
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name AS last_visited_animal FROM visits JOIN vets ON visits.vets_id = vets.id JOIN animals AS a ON visits.animals_id = a.id WHERE vets.name = 'William Tatcher' ORDER BY visit_date DESC LIMIT 1;
+
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(s.name) AS visit_different_animals, vets.name AS doctor_name FROM specializations JOIN vets ON specializations.vets_id = vets.id JOIN species AS s ON specializations.species_id = s.id WHERE vets.name = 'Stephanie Mendez' GROUP BY vets.name;
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT v.name AS vet_name, s.name AS speciliz_in FROM specializations AS sp FULL JOIN vets AS v ON sp.vets_id = v.id FULL JOIN species AS s ON sp.species_id = s.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animals_name, vets.name AS vets_name FROM visits AS v JOIN vets ON v.vets_id = vets.id JOIN animals AS a ON v.animals_id = a.id WHERE vets.name = 'Stephanie Mendez' AND visit_date BETWEEN '2020-04-01' AND '2020-08-31';
+
+-- What animal has the most visits to vets?
+SELECT COUNT(a.name) AS ttl_visits, a.name AS anim_name FROM visits AS v JOIN animals AS a ON v.animals_id = a.id GROUP BY a.name ORDER BY ttl_visits DESC LIMIT 1 ;
+
+-- Who was Maisy Smith's first visit?
+SELECT MIN(visit_date) AS first_visit, a.name AS anim_name FROM visits AS v JOIN vets AS vt ON v.vets_id = vt.id JOIN animals AS a ON v.animals_id = a.id WHERE vt.name = 'Maisy Smith' GROUP BY a.name ORDER BY first_visit ASC LIMIT 1;
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT MAX(visit_date) AS recent_visit, a.name AS anim_name, vt.name AS vet_name FROM visits AS v JOIN animals AS a ON v.animals_id = a.id JOIN vets AS vt ON v.vets_id = vt.id GROUP BY a.name, vt.name ORDER BY recent_visit DESC LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(visit_date) AS num_visit_with_unspecialize_vet FROM specializations AS sp JOIN vets AS vt ON sp.vets_id = vt.id JOIN visits AS v ON sp.vets_id = v.vets_id WHERE species_id IS NULL;
```

Comment:
- [ ] In this query you're listing the number of visits with a vet that doesn't have any specialization, however, we are looking for all the visits with a vet that doesn't specialize in that species which should add up to 12.

Reply:
@mikethreels Thank you for your review, the code has been corrected accordingly and the result is 12 now.